src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Markup/StyleXamlParser.cs

```diff
@@ -712,14 +712,6 @@ public override void WriteElementStart(XamlElementStartNode xamlElementStartNode
                     _inEventSetter = true;
 #endif
                 }
-                else if ((depth == 2 && _setterElementEncountered) ||
```

Comment:
Can you please explain a bit what purpose was served with this piece of code and why it is no longer required.

Reply:
This was a [validation](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fdotnet%2Fwpf%2Fblob%2Fmain%2Fsrc%2FMicrosoft.DotNet.Wpf%2Fsrc%2FPresentationFramework%2FSystem%2FWindows%2FMarkup%2FStyleXamlParser.cs%23L715-L722&data=05%7C02%7Canjalisheel%40microsoft.com%7Ca332e20dec3a4f72ceb108dbfb3df63c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638380015783311080%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=0Kabs9n87Q8bEugftX8f4rciY1i6ZMWb3EJXKjT1VdM%3D&reserved=0) which asserts that "Content" property in "Setter" is at a node depth of 2 or 3.

Due to the presence of this check, if you were to simply annotate the Setter class with [ContentProperty("Value")], the XAML compiler will fail to parse the XAML and throws exception during build.